### PR TITLE
guile-modules: add vyp as maintainer to unmaintained guile modules

### PR DESCRIPTION
**Pull request by vyp**
###### Motivation for this change

When [packaging guile-fibers](https://github.com/NixOS/nixpkgs/pull/29727), I noticed that a lot of the guile modules are unmaintained, so I would be happy to (try to) maintain them.

I also took this opportunity to rename the guile modules attributes in pkgs/top-level/all-packages.nix to be consistent kebab case (vs camel case and snake case), because git grepping showed that the ones I renamed were relatively unreferenced throughout nixpkgs. In fact only g-wrap referenced one of them, and the other references were from other guile modules themselves. Still, this at least affects @taktoa and @amiloradovsky, so you guys would have to be okay with this? Also, maybe there is a policy against renaming top level attributes? Obviously I'm willing to revert this change if anyone has issues with it.

I also took the opportunity to make a lot of stylistic changes to the modules I've added myself as maintainer for, seeing as though I would be the sole maintainer (so I think no one should have issues with it).

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by taktoa** [comment]
I no longer use Guile, so I have no qualms with any changes to the Guile Nix packages (in fact, feel free to remove me as a package maintainer if you feel up to the task). Also, I did a bunch of work on packaging Guile libraries around 2 years ago; you may be interested in [what I wrote back then](https://github.com/taktoa/nix-packages/tree/47c751b8bda92821b6934c764772a5aaa6b5d448/guilePackages).

**Comment by vyp** [comment]
@taktoa Ok thanks for letting me know, and thanks so much for your previous contributions! :cake: And thanks for the link too!

I shall rebase this sometime.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -6708,18 +6708,18 @@ with pkgs;
 
   jython = callPackage ../development/interpreters/jython {};
```
Comment (Mic92) [inline pkgs/top-level/all-packages.nix]: Can you also add aliases to `pkgs/top-level/aliases.nix`?

Comment (vyp) [reply pkgs/top-level/all-packages.nix]: Nice, didn't know about that. Done.

**Comment by Mic92** [comment]
Can you take a look at `guile-gnome`. It fails to build for me:

```
guile-runtime.c:357:3: error: 'scm_class_generic' is deprecated [-Werror=deprecated-declarations]
   } else if (scm_is_true (scm_call_2 (is_a_p_proc, scm_variable_ref (var),
   ^
In file included from /nix/store/kb01d4sfmbay1bdwnvnay536jvimlxn4-guile-2.2.0-dev/include/guile/2.2/libguile.h:128:0,
                 from ../../guile/g-wrap/guile-compatibility.h:25,
                 from guile-runtime.c:32:
/nix/store/kb01d4sfmbay1bdwnvnay536jvimlxn4-guile-2.2.0-dev/include/guile/2.2/libguile/deprecated.h:174:20: note: declared here
 SCM_DEPRECATED SCM scm_class_generic;
                    ^~~~~~~~~~~~~~~~~
guile-runtime.c:367:28: error: 'scm_class_generic' is deprecated [-Werror=deprecated-declarations]
                            scm_list_5 (scm_class_generic,
                            ^~~~~~~~~~
In file included from /nix/store/kb01d4sfmbay1bdwnvnay536jvimlxn4-guile-2.2.0-dev/include/guile/2.2/libguile.h:128:0,
                 from ../../guile/g-wrap/guile-compatibility.h:25,
                 from guile-runtime.c:32:
/nix/store/kb01d4sfmbay1bdwnvnay536jvimlxn4-guile-2.2.0-dev/include/guile/2.2/libguile/deprecated.h:174:20: note: declared here
 SCM_DEPRECATED SCM scm_class_generic;
                    ^~~~~~~~~~~~~~~~~
guile-runtime.c: In function 'gw_guile_handle_wrapper_error':
guile-runtime.c:759:7: error: 'scm_memory_error' is deprecated [-Werror=deprecated-declarations]
       scm_memory_error(func_name);
       ^~~~~~~~~~~~~~~~
In file included from /nix/store/kb01d4sfmbay1bdwnvnay536jvimlxn4-guile-2.2.0-dev/include/guile/2.2/libguile.h:128:0,
                 from ../../guile/g-wrap/guile-compatibility.h:25,
                 from guile-runtime.c:32:
/nix/store/kb01d4sfmbay1bdwnvnay536jvimlxn4-guile-2.2.0-dev/include/guile/2.2/libguile/deprecated.h:148:21: note: declared here
 SCM_DEPRECATED void scm_memory_error (const char *subr) SCM_NORETURN;
                     ^~~~~~~~~~~~~~~~
cc1: all warnings being treated as errors
make[4]: *** [Makefile:596: guile-runtime.lo] Error 1
```

I looks like it could be solved by disabling `-Werror=deprecated-declarations

**Comment by vyp** [comment]
Or could probably just update guile-gnome to 2.16.5 which is apparently compatible with guile 2.2. In any case I'm not maintaining that one, I just changed some of the variables because I didn't know about pkgs/top-level/aliases.nix, and furthermore this PR was just supposed to mainly be for adding me as maintainer. I think there's at least another module that could do with an update, but I wanted to do it separately (in case of complications/build failures).

Although I am willing to maintain gnome-guile as well if I wanted, and/or make a PR to update it to 2.16.5.

**Comment by amiloradovsky** [comment]
@vyp I'm actually supportive of this renaming. Consistency improvements are welcome.
And, yes, if you'd fix/update gnome-guile, that would be nice too. (Since something like `NIX_CFLAGS_COMPILE = "-Wno-deprecated-declarations";` isn't really the way to go.)
Unfortunately, I myself am not interested in the GNOME bindings anymore, so you could remove me from the maintainers list, by the way.

**Comment by vyp** [comment]
@amiloradovsky Cool. :tada:

But okay, I guess I shall remove both you and @taktoa from the maintainers for guile-gnome and add myself to it sometime. And also update it.

**Pull request merged**